From patchwork Thu Dec 12 02:45:10 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Gavin Shan <gshan@redhat.com>
X-Patchwork-Id: 11286801
Return-Path: <SRS0=BLLP=2C=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 02F466C1
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 12 Dec 2019 02:45:32 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id CBEF521655
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 12 Dec 2019 02:45:31 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="KDRzkLEh"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727882AbfLLCpb (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 11 Dec 2019 21:45:31 -0500
Received: from us-smtp-1.mimecast.com ([205.139.110.61]:60788 "EHLO
        us-smtp-delivery-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL)
        by vger.kernel.org with ESMTP id S1727879AbfLLCpa (ORCPT
        <rfc822;kvm@vger.kernel.org>); Wed, 11 Dec 2019 21:45:30 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1576118729;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:
         content-transfer-encoding:content-transfer-encoding;
        bh=vVBnTdpElxP7oPw8MW4U3TzLCcSf5UiwUt7vzIeW4ps=;
        b=KDRzkLEhJj0gaQt9UdggpFq4eXkoZaPWQl/rrfiphS+g6gBsXpC/Jhx7GagDAyFzxit9CW
        FG1HWoMgbCTsmTd2UkwRvwPGy9qCGkoQFF5QpB4tpr9FTVGWpbe+Ts2l+lH5ImeLBBzwQQ
        MFYNS/SV9zMPqL3tNssOUMA6hLC6PEc=
Received: from mimecast-mx01.redhat.com (mimecast-mx01.redhat.com
 [209.132.183.4]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-282-G0VUWj9PM5qw6TyRw48vBw-1; Wed, 11 Dec 2019 21:45:28 -0500
X-MC-Unique: G0VUWj9PM5qw6TyRw48vBw-1
Received: from smtp.corp.redhat.com (int-mx03.intmail.prod.int.phx2.redhat.com
 [10.5.11.13])
        (using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
        (No client certificate requested)
        by mimecast-mx01.redhat.com (Postfix) with ESMTPS id 8142F800D4C;
        Thu, 12 Dec 2019 02:45:27 +0000 (UTC)
Received: from localhost.localdomain.com (vpn2-54-40.bne.redhat.com
 [10.64.54.40])
        by smtp.corp.redhat.com (Postfix) with ESMTP id 699A46609A;
        Thu, 12 Dec 2019 02:45:24 +0000 (UTC)
From: Gavin Shan <gshan@redhat.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, rkrcmar@redhat.com, paulus@ozlabs.org,
        maz@kernel.org, jhogan@kernel.org, drjones@redhat.com,
        vkuznets@redhat.com, gshan@redhat.com
Subject: [PATCH 1/3] kvm/mips: Standardize kvm exit reason field
Date: Thu, 12 Dec 2019 13:45:10 +1100
Message-Id: <20191212024512.39930-2-gshan@redhat.com>
MIME-Version: 1.0
X-Scanned-By: MIMEDefang 2.79 on 10.5.11.13
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

This standardizes kvm exit reason field name by replacing "reason"
with "exit_reason".

Signed-off-by: Gavin Shan <gshan@redhat.com>
---
 arch/mips/kvm/trace.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/mips/kvm/trace.h b/arch/mips/kvm/trace.h
index a8c7fd7bf6d2..acbbe3fff9de 100644
--- a/arch/mips/kvm/trace.h
+++ b/arch/mips/kvm/trace.h
@@ -113,20 +113,20 @@ DEFINE_EVENT(kvm_transition, kvm_out,
 	{ KVM_TRACE_EXIT_GPA,		"GPA" }
 
 TRACE_EVENT(kvm_exit,
-	    TP_PROTO(struct kvm_vcpu *vcpu, unsigned int reason),
-	    TP_ARGS(vcpu, reason),
+	    TP_PROTO(struct kvm_vcpu *vcpu, unsigned int exit_reason),
+	    TP_ARGS(vcpu, exit_reason),
 	    TP_STRUCT__entry(
 			__field(unsigned long, pc)
-			__field(unsigned int, reason)
+			__field(unsigned int, exit_reason)
 	    ),
 
 	    TP_fast_assign(
 			__entry->pc = vcpu->arch.pc;
-			__entry->reason = reason;
+			__entry->exit_reason = exit_reason;
 	    ),
 
 	    TP_printk("[%s]PC: 0x%08lx",
-		      __print_symbolic(__entry->reason,
+		      __print_symbolic(__entry->exit_reason,
 				       kvm_trace_symbol_exit_types),
 		      __entry->pc)
 );

From patchwork Thu Dec 12 02:45:11 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Gavin Shan <gshan@redhat.com>
X-Patchwork-Id: 11286803
Return-Path: <SRS0=BLLP=2C=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id A8E2F6C1
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 12 Dec 2019 02:45:36 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 880DB22527
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 12 Dec 2019 02:45:36 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="HU4uXUbF"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727879AbfLLCpf (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 11 Dec 2019 21:45:35 -0500
Received: from us-smtp-2.mimecast.com ([207.211.31.81]:36044 "EHLO
        us-smtp-delivery-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL)
        by vger.kernel.org with ESMTP id S1727756AbfLLCpf (ORCPT
        <rfc822;kvm@vger.kernel.org>); Wed, 11 Dec 2019 21:45:35 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1576118734;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:
         content-transfer-encoding:content-transfer-encoding;
        bh=sr810Oqrdalkpz1AS7AmLIEu53E2Nn5hxhJYZnRzfmA=;
        b=HU4uXUbFKYFYGdV0Y02Qy2TPPRThdhYMrQGsKFfDiM4fgrkeKdx98TS+7MrJs7p9K1NZaz
        7+6COUj8Okzhkk44w7zKzSo25ueXKGnVsMZMKCWwakpOnRaOOxjiaC+z6ry3VMu+UHD4NP
        Cap8QpCtx1KLWVe1eLZgJirq1soV9C8=
Received: from mimecast-mx01.redhat.com (mimecast-mx01.redhat.com
 [209.132.183.4]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-217-FeCsVO4jN9Oa1C8NMATUAw-1; Wed, 11 Dec 2019 21:45:33 -0500
X-MC-Unique: FeCsVO4jN9Oa1C8NMATUAw-1
Received: from smtp.corp.redhat.com (int-mx03.intmail.prod.int.phx2.redhat.com
 [10.5.11.13])
        (using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
        (No client certificate requested)
        by mimecast-mx01.redhat.com (Postfix) with ESMTPS id 3CCFE107ACCD;
        Thu, 12 Dec 2019 02:45:32 +0000 (UTC)
Received: from localhost.localdomain.com (vpn2-54-40.bne.redhat.com
 [10.64.54.40])
        by smtp.corp.redhat.com (Postfix) with ESMTP id 7144276FFF;
        Thu, 12 Dec 2019 02:45:29 +0000 (UTC)
From: Gavin Shan <gshan@redhat.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, rkrcmar@redhat.com, paulus@ozlabs.org,
        maz@kernel.org, jhogan@kernel.org, drjones@redhat.com,
        vkuznets@redhat.com, gshan@redhat.com
Subject: [PATCH 2/3] kvm/powerpc: Standardize kvm exit reason field
Date: Thu, 12 Dec 2019 13:45:11 +1100
Message-Id: <20191212024512.39930-3-gshan@redhat.com>
MIME-Version: 1.0
X-Scanned-By: MIMEDefang 2.79 on 10.5.11.13
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

This standardizes kvm exit reason field name by replacing "exit_nr"
with "exit_reason".

Signed-off-by: Gavin Shan <gshan@redhat.com>
---
 arch/powerpc/kvm/trace_booke.h | 10 +++++-----
 arch/powerpc/kvm/trace_pr.h    | 10 +++++-----
 2 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/arch/powerpc/kvm/trace_booke.h b/arch/powerpc/kvm/trace_booke.h
index 3837842986aa..f757ce1dccb9 100644
--- a/arch/powerpc/kvm/trace_booke.h
+++ b/arch/powerpc/kvm/trace_booke.h
@@ -36,11 +36,11 @@
 	{41, "HV_PRIV"}
 
 TRACE_EVENT(kvm_exit,
-	TP_PROTO(unsigned int exit_nr, struct kvm_vcpu *vcpu),
-	TP_ARGS(exit_nr, vcpu),
+	TP_PROTO(unsigned int exit_reason, struct kvm_vcpu *vcpu),
+	TP_ARGS(exit_reason, vcpu),
 
 	TP_STRUCT__entry(
-		__field(	unsigned int,	exit_nr		)
+		__field(	unsigned int,	exit_reason	)
 		__field(	unsigned long,	pc		)
 		__field(	unsigned long,	msr		)
 		__field(	unsigned long,	dar		)
@@ -48,7 +48,7 @@ TRACE_EVENT(kvm_exit,
 	),
 
 	TP_fast_assign(
-		__entry->exit_nr	= exit_nr;
+		__entry->exit_reason	= exit_reason;
 		__entry->pc		= kvmppc_get_pc(vcpu);
 		__entry->dar		= kvmppc_get_fault_dar(vcpu);
 		__entry->msr		= vcpu->arch.shared->msr;
@@ -61,7 +61,7 @@ TRACE_EVENT(kvm_exit,
 		" | dar=0x%lx"
 		" | last_inst=0x%lx"
 		,
-		__print_symbolic(__entry->exit_nr, kvm_trace_symbol_exit),
+		__print_symbolic(__entry->exit_reason, kvm_trace_symbol_exit),
 		__entry->pc,
 		__entry->msr,
 		__entry->dar,
diff --git a/arch/powerpc/kvm/trace_pr.h b/arch/powerpc/kvm/trace_pr.h
index 46a46d328fbf..b6039777e2b4 100644
--- a/arch/powerpc/kvm/trace_pr.h
+++ b/arch/powerpc/kvm/trace_pr.h
@@ -215,11 +215,11 @@ TRACE_EVENT(kvm_book3s_slbmte,
 );
 
 TRACE_EVENT(kvm_exit,
-	TP_PROTO(unsigned int exit_nr, struct kvm_vcpu *vcpu),
-	TP_ARGS(exit_nr, vcpu),
+	TP_PROTO(unsigned int exit_reason, struct kvm_vcpu *vcpu),
+	TP_ARGS(exit_reason, vcpu),
 
 	TP_STRUCT__entry(
-		__field(	unsigned int,	exit_nr		)
+		__field(	unsigned int,	exit_reason	)
 		__field(	unsigned long,	pc		)
 		__field(	unsigned long,	msr		)
 		__field(	unsigned long,	dar		)
@@ -228,7 +228,7 @@ TRACE_EVENT(kvm_exit,
 	),
 
 	TP_fast_assign(
-		__entry->exit_nr	= exit_nr;
+		__entry->exit_reason	= exit_reason;
 		__entry->pc		= kvmppc_get_pc(vcpu);
 		__entry->dar		= kvmppc_get_fault_dar(vcpu);
 		__entry->msr		= kvmppc_get_msr(vcpu);
@@ -243,7 +243,7 @@ TRACE_EVENT(kvm_exit,
 		" | srr1=0x%lx"
 		" | last_inst=0x%lx"
 		,
-		__print_symbolic(__entry->exit_nr, kvm_trace_symbol_exit),
+		__print_symbolic(__entry->exit_reason, kvm_trace_symbol_exit),
 		__entry->pc,
 		__entry->msr,
 		__entry->dar,

From patchwork Thu Dec 12 02:45:12 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Gavin Shan <gshan@redhat.com>
X-Patchwork-Id: 11286805
Return-Path: <SRS0=BLLP=2C=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id AA5586C1
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 12 Dec 2019 02:45:40 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 802B62464B
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 12 Dec 2019 02:45:40 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="H1vUJz6z"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727893AbfLLCpj (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 11 Dec 2019 21:45:39 -0500
Received: from us-smtp-1.mimecast.com ([207.211.31.81]:38816 "EHLO
        us-smtp-delivery-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL)
        by vger.kernel.org with ESMTP id S1727892AbfLLCpj (ORCPT
        <rfc822;kvm@vger.kernel.org>); Wed, 11 Dec 2019 21:45:39 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1576118739;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:
         content-transfer-encoding:content-transfer-encoding;
        bh=uncFyvnHFSKVFSYnpKTKrWHTgCtzsOrAlALXlYSzS3Q=;
        b=H1vUJz6zbzdTtXYfq4vq7Bmx57LIKVKMjE+XxNi5n7CuHLyKW2vDTHpk3UK0sC38rsRcXl
        3ZL1hssks3KCOcAZ3i/t+iwBEFKgdywQozZDgt0nDmfsF3Cue0iX5zir5F4rIYAIr/37EC
        R/GDVR7pHrdDdpTOBaDJyEKd0dtqgYo=
Received: from mimecast-mx01.redhat.com (mimecast-mx01.redhat.com
 [209.132.183.4]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-159-22lQ5p1LPcueCQXL0rT7Ew-1; Wed, 11 Dec 2019 21:45:38 -0500
X-MC-Unique: 22lQ5p1LPcueCQXL0rT7Ew-1
Received: from smtp.corp.redhat.com (int-mx03.intmail.prod.int.phx2.redhat.com
 [10.5.11.13])
        (using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
        (No client certificate requested)
        by mimecast-mx01.redhat.com (Postfix) with ESMTPS id CE2BC1800D45;
        Thu, 12 Dec 2019 02:45:36 +0000 (UTC)
Received: from localhost.localdomain.com (vpn2-54-40.bne.redhat.com
 [10.64.54.40])
        by smtp.corp.redhat.com (Postfix) with ESMTP id 9CED16FEF8;
        Thu, 12 Dec 2019 02:45:33 +0000 (UTC)
From: Gavin Shan <gshan@redhat.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, rkrcmar@redhat.com, paulus@ozlabs.org,
        maz@kernel.org, jhogan@kernel.org, drjones@redhat.com,
        vkuznets@redhat.com, gshan@redhat.com
Subject: [PATCH 3/3] kvm/arm: Standardize kvm exit reason field
Date: Thu, 12 Dec 2019 13:45:12 +1100
Message-Id: <20191212024512.39930-4-gshan@redhat.com>
MIME-Version: 1.0
X-Scanned-By: MIMEDefang 2.79 on 10.5.11.13
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

This standardizes kvm exit reason field name by replacing "esr_ec"
with "exit_reason".

Signed-off-by: Gavin Shan <gshan@redhat.com>
---
 virt/kvm/arm/trace.h | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/virt/kvm/arm/trace.h b/virt/kvm/arm/trace.h
index 204d210d01c2..0ac774fd324d 100644
--- a/virt/kvm/arm/trace.h
+++ b/virt/kvm/arm/trace.h
@@ -27,25 +27,27 @@ TRACE_EVENT(kvm_entry,
 );
 
 TRACE_EVENT(kvm_exit,
-	TP_PROTO(int ret, unsigned int esr_ec, unsigned long vcpu_pc),
-	TP_ARGS(ret, esr_ec, vcpu_pc),
+	TP_PROTO(int ret, unsigned int exit_reason, unsigned long vcpu_pc),
+	TP_ARGS(ret, exit_reason, vcpu_pc),
 
 	TP_STRUCT__entry(
 		__field(	int,		ret		)
-		__field(	unsigned int,	esr_ec		)
+		__field(	unsigned int,	exit_reason	)
 		__field(	unsigned long,	vcpu_pc		)
 	),
 
 	TP_fast_assign(
 		__entry->ret			= ARM_EXCEPTION_CODE(ret);
-		__entry->esr_ec = ARM_EXCEPTION_IS_TRAP(ret) ? esr_ec : 0;
+		__entry->exit_reason =
+			ARM_EXCEPTION_IS_TRAP(ret) ? exit_reason: 0;
 		__entry->vcpu_pc		= vcpu_pc;
 	),
 
 	TP_printk("%s: HSR_EC: 0x%04x (%s), PC: 0x%08lx",
 		  __print_symbolic(__entry->ret, kvm_arm_exception_type),
-		  __entry->esr_ec,
-		  __print_symbolic(__entry->esr_ec, kvm_arm_exception_class),
+		  __entry->exit_reason,
+		  __print_symbolic(__entry->exit_reason,
+				   kvm_arm_exception_class),
 		  __entry->vcpu_pc)
 );
 
